src/client/datascience/raw-kernel/rawSession.ts

```diff
@@ -39,6 +39,19 @@ export class RawSession implements ISessionWithSocket {
     private readonly _connectionStatusChanged: Signal<this, Kernel.ConnectionStatus>;
     private readonly exitHandler: IDisposable;
     private readonly signaling: typeof import('@lumino/signaling');
+    private _jupyterLabServices?: typeof import('@jupyterlab/services')
+    private cellExecutedSuccessfully?: boolean;
+    public get AtleastOneCellExecutedSuccessfully() {
```

Comment:
Damn it, was meant for another PR, but leaving this.
Basically this allows me to tell whether we successfully executed at least one cell (user code) against this session/kernel.

This is required to implement the logic we discussed (restart only if we have successfully run a cell ,etc...)
that stuff is in a separate PR.